mp2-common/src/mpt_sequential/leaf_or_extension.rs

```diff
@@ -70,19 +81,78 @@ impl MPTLeafOrExtensionNode {
 
         // Advance the key and extract the value (only decode two headers in the case of leaf).
         let rlp_headers = decode_fixed_list::<_, D, 2>(b, &node.arr.arr, zero);
-        let (key, value, valid) = MPTCircuit::<1, NODE_LEN>::advance_key_leaf_or_extension::<
-            F,
-            D,
-            2,
-            VALUE_LEN,
-        >(b, &node.arr, current_key, &rlp_headers);
+        let (key, value, valid) =
+            advance_key_leaf_or_extension::<F, D, 2, VALUE_LEN, NODE_LEN, KEY_LEN>(
+                b,
+                &node.arr,
+                current_key,
+                &rlp_headers,
+            );
         b.connect(tru.target, valid.target);
 
-        MPTLeafOrExtensionWires {
+        MPTLeafOrExtensionWiresGeneric {
             node,
             root,
             key,
             value,
         }
     }
 }
+
+/// Wrapped wires for a MPT receipt leaf
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MPTReceiptLeafWiresGeneric<const NODE_LEN: usize, const KEY_LEN: usize>
```

Comment:
This is gonna be the same for the transaction trie, so maybe `IndexMPTKey` or stg ? to say that the index is now the key.

Reply:
Can you put some comments about the KEY_LEN,i.e. what should be a good range for example.